CMakeLists.txt

```diff
@@ -26,24 +48,49 @@ target_include_directories(${PROJECT_NAME}
             $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>/External/tinyxml2
 )
 
+# Windows-specific linking
 if(WIN32)
     target_link_libraries(${PROJECT_NAME}
         PUBLIC 
-            $<$<STREQUAL:"${CMAKE_CXX_COMPILER_ID}","MSVC">:ws2_32.lib iphlpapi.lib>
-            $<$<STREQUAL:"${CMAKE_CXX_COMPILER_ID}","GNU">:ws2_32 iphlpapi>
+            ws2_32 iphlpapi
     )
 endif()
```

Comment:
suggestion: Remove `if(WIN32)` and use [cmake-generator-expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html)

Something like
`$<$<BOOL:${WIN32}>>:ws2_32>`
`$<$<BOOL:${WIN32}>>:iphlpapi>`

Reply:
Fixed.